src/script/ismine.h

```diff
@@ -9,22 +9,48 @@
 #include <script/standard.h>
 
 #include <stdint.h>
+#include <bitset>
 
 class CKeyStore;
 class CScript;
 
 /** IsMine() return codes */
 enum isminetype
 {
-    ISMINE_NO = 0,
-    ISMINE_WATCH_ONLY = 1,
-    ISMINE_SPENDABLE = 2,
-    ISMINE_ALL = ISMINE_WATCH_ONLY | ISMINE_SPENDABLE
+    ISMINE_NO         = 0,
+    ISMINE_WATCH_ONLY = 1 << 0,
+    ISMINE_SPENDABLE  = 1 << 1,
+    ISMINE_ALL        = ISMINE_WATCH_ONLY | ISMINE_SPENDABLE,
+    ISMINE_ENUM_ELEMENTS,
 };
 /** used for bitflags of isminetype */
 typedef uint8_t isminefilter;
 
 isminetype IsMine(const CKeyStore& keystore, const CScript& scriptPubKey);
 isminetype IsMine(const CKeyStore& keystore, const CTxDestination& dest);
 
+struct CachableAccount
+{
+    enum Type {
+        CREDIT,
+        DEBIT,
+    } m_type{CREDIT};
+    std::bitset<ISMINE_ENUM_ELEMENTS> m_cached;
+    CAmount m_value[ISMINE_ENUM_ELEMENTS];
+    CachableAccount(Type type = CREDIT) : m_type(type)
+    {
+        Reset();
+    }
+    void Reset()
+    {
+        m_cached.reset();
+    }
+    CachableAccount& Set(const isminefilter& filter, CAmount value)
```

Comment:
I don't see a reason to pass enum values by const reference? (despite the fact there's some occurrences in the code).

Reply:
I don't see the return value being used so I'd drop it.